File: sdk/samples/WPFSamples/SafeFileUploadPartialTrustSample/visualbasic/imagecustomcontentstate.vb
Imports System
Imports System.Windows.Media
Imports System.Windows.Navigation

<Serializable()> _
    Public Class ImageCustomContentState
    Inherits CustomContentState

    Public Sub New(ByVal imageSource As ImageSource, ByVal filename As String)
        Me.imageSource = imageSource
        Me.filename = filename
    End Sub

    Public Overrides Sub Replay(ByVal navigationService As NavigationService, ByVal mode As NavigationMode)
        Dim homePage As HomePage = DirectCast(navigationService.Content, HomePage)
        homePage.viewImage.Source = Me.imageSource
        homePage.nameLabel.Content = Me.filename
    End Sub

    Public Overrides ReadOnly Property JournalEntryName() As String
        Get
            Return Me.filename
        End Get
    End Property

    Private filename As String
    Private imageSource As ImageSource

End Class

File: sdk/samples/WPFSamples/SafeFileUploadPartialTrustSample/csharp/imagecustomcontentstate.cs
using System;
using System.IO;
using System.Windows;
using System.Windows.Media;
using System.Windows.Media.Imaging;
using System.Windows.Navigation;

namespace SafeFileUploadPartialTrustSample
{

    [Serializable]
    public class ImageCustomContentState : CustomContentState
    {

        ImageSource imageSource;
        string filename;

        public ImageCustomContentState(ImageSource imageSource, string filename)
        {
            this.imageSource = imageSource;
            this.filename = filename;
        }

        public override string JournalEntryName
        {
            get { return this.filename; }
        }

        public override void Replay(NavigationService navigationService, NavigationMode mode)
        {
            HomePage homePage = (HomePage)navigationService.Content;
            homePage.viewImage.Source = this.imageSource;
            homePage.nameLabel.Content = this.filename;
        }
    }
}
